app/controllers/api/v1/listen_signals_controller.rb

```diff
@@ -96,16 +100,30 @@ def listen_signal_template_types
     }
   end
 
+  def patch_update
+    @listen_signal.update!(patch_signal_params)
+  end
+
   def put_update
     signal_params = update_signal_params
-    @listen_signal.update!(signal_params)
+    response_params = create_response_params
 
-    update_response(@listen_signal.default_response, params[:default_response])
-    update_response(@listen_signal.repeat_response, params[:repeat_response])
+    ActiveRecord::Base.transaction do
```

Comment:
This controller is getting bloated. Can we move the create and update listen signal logic to a service?